src/main/java/org/apache/commons/io/input/QueueInputStream.java

```diff
@@ -88,12 +111,32 @@ public QueueOutputStream newQueueOutputStream() {
     /**
      * Reads and returns a single byte.
      *
-     * @return either the byte read or {@code -1} if the end of the stream has been reached
+     * @return either the byte read or {@code -1} immediately if the queue is empty.
      */
     @Override
     public int read() {
         final Integer value = blockingQueue.poll();
         return value == null ? EOF : 0xFF & value;
     }
 
+    private static class BlockingQueueInputStream extends QueueInputStream {
+
+        public BlockingQueueInputStream(final BlockingQueue<Integer> blockingQueue) {
+            super(blockingQueue);
+        }
+
+        @Override
+        public int read() {
+            try {
+                return blockingQueue.take();
+            } catch (final InterruptedException e) {
+                Thread.currentThread().interrupt();
+                final InterruptedIOException ioException = new InterruptedIOException();
```

Comment:
@maxxedev 
This seems unusual, why not simply use `new InterruptedIOException(e)` or even simpler `throw new UncheckedIOException(e)`?

Reply:
UncheckedIOException constructor requires a IOException.

InterruptedException does not extend IOException
InterruptedIOException extends IOException